ppa/archive/views.py

```diff
@@ -166,3 +170,73 @@ def render_to_csv(self, data):
 
     def get(self, *args, **kwargs):
         return self.render_to_csv(self.get_data())
+
+
+class BulkAddCollectionView(ListView, FormMixin, ProcessFormView):
+    '''
+    View to bulk add a queryset of :class:ppa.archive.models.DigitizedWork
+    to a set of :class:ppa.archive.models.Collection instances.
+
+    Restricted to staff users via staff_member_required on url.
+    '''
+
+    model = DigitizedWork
+    template_name = 'admin/archive/digitizedwork/bulk_add_collections.html'
+    form_class = BulkAddCollectionForm
+    success_url = reverse_lazy('admin:archive_digitizedwork_changelist')
+
+    def get_queryset(self, *args, **kwargs):
+        '''Return a queryset filtered by id, or empty list if no ids'''
+        ids = self.request.GET.get('ids', None)
+        # if somehow a problematic non-pk is pushed, will be ignored in filter
+        return DigitizedWork.objects.filter(id__in=ids.split(',')
+                                            if ids else [])
+
+    def get_initial(self):
+        '''
+        Set the hidden field with the :class:ppa.archive.models.DigitizedWork
+        to add to collection(s) to pass to POST request.
+        '''
+        initial = super(BulkAddCollectionView, self).get_initial()
+        # this ensures that the hidden field for POST only ever contains
+        # actual PKs
+        ids = ','.join(str(val) for val in
+                       self.get_queryset().values_list('id', flat=True))
+        if ids:
+            initial['digitized_work_ids'] = ids
+        return initial
+
+    def post(self, request, *args, **kwargs):
+        '''
+        Add :class:ppa.archive.models.DigitizedWork instances passed in form
+        data to selected instances of :class:ppa.archive.models.Collection,
```

Comment:
I think your `:class:` docstring syntax isn't quite right (don't you need backticks around class names?).  You might want to build the sphinx doc and check.

Reply:
I did indeed miss those. I'll double check building the sphinx doc before finishing work on revisions, but I believe I edited most of the instances of this while refactoring.